api/v1alpha4/vspheredeploymentzone_webhook.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha4
+
+import (
+	"k8s.io/utils/pointer"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+func (r *VSphereDeploymentZone) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/mutate-infrastructure-cluster-x-k8s-io-v1alpha4-vspheredeploymentzone,mutating=true,failurePolicy=fail,groups=infrastructure.cluster.x-k8s.io,resources=vspheredeploymentzones,verbs=create;update,versions=v1alpha4,name=validation.vspheredeploymentzone.infrastructure.x-k8s.io,sideEffects=None,admissionReviewVersions=v1beta1
+
+var _ webhook.Defaulter = &VSphereDeploymentZone{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *VSphereDeploymentZone) Default() {
+	if r.Spec.ControlPlane == nil {
```

Comment:
is this different from setting `kubebuilder:default`

Reply:
Not sure if this can be done via the annotation since this is a pointer, can try this out.